pkg/flowaggregator/clickhouseclient/clickhouseclient.go

```diff
@@ -518,3 +512,84 @@ func (ch *ClickHouseExportProcess) batchCommitAll() (int, error) {
 
 	return currSize, nil
 }
+
+func CheckConnection(input ClickHouseInput) (string, *sql.DB, error) {
+	dsn, err := input.getDataSourceName()
+	if err != nil {
+		klog.ErrorS(err, "Error parsing ClickHouse DSN")
+		return "", nil, err
+	}
+	connect, err := sql.Open("clickhouse", dsn)
+	if err != nil {
+		klog.ErrorS(err, "Error when opening DB connection", "dbURL", input.DatabaseURL)
+		return "", nil, err
+	}
+
+	// Test ping DB
+	if err = connect.Ping(); err != nil {
+		if exception, ok := err.(*clickhouse.Exception); ok {
+			klog.ErrorS(err, "Error pinging DB", "errCode", exception.Code)
+		} else {
+			klog.ErrorS(err, "Error pinging DB")
+		}
+		return "", nil, err
+	}
+	// Test open Transaction
+	tx, err := connect.Begin()
+	if err == nil {
+		_, err = tx.Prepare(insertQuery)
+	}
+	if err != nil {
+		klog.ErrorS(err, "Error when preparing insert statement")
+		return "", nil, err
+	}
+	_ = tx.Commit()
+	return dsn, connect, err
+}
+
+func (ch *ClickHouseExportProcess) GetDsnMap() map[string]string {
+	parseURL := strings.Split(ch.dsn, "?")
+	m := make(map[string]string)
+	m["databaseURL"] = parseURL[0]
+	for _, v := range strings.Split(parseURL[1], "&") {
+		pair := strings.Split(v, "=")
+		m[pair[0]] = pair[1]
+	}
+	return m
+}
+
+func (ch *ClickHouseExportProcess) UpdateCH(input *ClickHouseExportProcess, dsn string, connect *sql.DB) {
+	input.mutex.Lock()
+	defer input.mutex.Unlock()
+	input.dsn = dsn
+	input.db = connect
+}
+
+func (ch *ClickHouseExportProcess) GetCommitInterval() time.Duration {
+	return ch.commitInterval
+}
+
+func (ch *ClickHouseExportProcess) SetCommitInterval(commitInterval time.Duration) {
+	ch.commitInterval = commitInterval
+	ch.commitTicker.Reset(ch.commitInterval)
+}
+
+func (ch *ClickHouseExportProcess) UpdateTTL(ttl string) error {
+
+	query := "ALTER TABLE flows MODIFY TTL timeInserted + INTERVAL (?) SECOND"
```

Comment:
You may also want to update `merge_with_ttl_timeout` with something like `ALTER TABLE flows MODIFY SETTING merge_with_ttl_timeout=100`. IMO, it would be good to set to min(TTL, 14400).

Reply:
May I know the reason why we need a maximum for merge_with_ttl_timeout?

Reply:
It's like when user set TTL to months, we still would like to have `merge_with_ttl_timeout` in hours level so that the deleting will be "continuous" instead of having a deletion of tons of records every several months. Propose 14400 as it is the default value given by ClickHouse.

Reply:
I know what you're saying. min(TTL, merge_with_ttl_timeout) is used to make sure our db is upadted.

Reply:
I've changed it. Thanks.